contrib/storage-splunk/src/main/java/org/apache/drill/exec/store/splunk/SplunkGroupScan.java

```diff
@@ -50,17 +54,20 @@ public class SplunkGroupScan extends AbstractGroupScan {
 
   private int hashCode;
 
+  private MetadataProviderManager metadataProviderManager;
```

Comment:
Can you elaborate on the role played by the metastore in this PR a bit, please?

Reply:
This actually isn't related to the metastore.  It is some boilerplate code to allow for provided schema.  You have to have  the MetadataProviderManager to access the inline schemata, which you then add to the schema negotiator in the ScanBatchCreator class.  This also is why I added all of the `getPhysicalScan()` functions to the `SplunkStoragePlugin` class.   With that said, this PR does not go all the way and add provided schema support.  I will do that as a separate PR.

For future storage plugins, we should make that a standard, as it really isn't difficult to add.